snews_cs/cs_email.py

```diff
@@ -24,26 +37,46 @@ def send_email(alert_content):
     for k, v in alert_content.items():
         space = 40 - len(k)
         pretty_alert += f'{k} :{r" "*space}{v}\n'
+    subject = "SNEWS COINCIDENCE" + datetime.utcnow().isoformat()
     emails = 'snews2-test-ahabig@d.umn.edu'
-    os.system( f'echo "{pretty_alert}"| mail -s "SNEWS COINCIDENCE {datetime.utcnow().isoformat()}" {emails}')
+
+    _smtp_sender(pretty_alert, subject, emails)
     log.info(f"\t\t> SNEWS Alert mail was sent at {datetime.utcnow().isoformat()} to {emails}")
 
-
-def _mail_sender(mails):
-    """ Send the generated emails via s-nail
-    """
-    success = False
-    for i, mail in enumerate(mails[::-1]):
-        # try sending either of the mails
-        stdout = os.system(mail)
-        if stdout == 0:
-            success = True
-    if success:
-        log.info(f"\t> Mail was successfully sent!\n")
-        return True
-    else:
-        log.error(f"\t> Mail could not be sent.\n")
-        return False
+### _smtp_sender service function.  All other functions in this file 
+### call this for mail handling.
+def _smtp_sender(body, subject, addr, attachment=None):
+    # Create a text/plain message
+    msg = email.mime.multipart.MIMEMultipart()
+    msg['Subject'] = subject
+
+    msg['From'] = 'SNEWS USER <' + sender + '>'
+    msg['To'] = addr
+
+    # The main body is just another attachment
+    emailbody = email.mime.text.MIMEText(body)
+    msg.attach(emailbody)
+
+    if attachment is not None:
+         fp=open(attachment,'rb')
+         att = email.mime.application.MIMEApplication(fp.read(),_subtype="octet-stream")
+         fp.close()
+         att.add_header('Content-Disposition','attachment',filename=attachment)
+         msg.attach(att)
+
+    # TODO: This should be defined as a os env
+    with smtplib.SMTP(smtpserver) as smtp:
+         smtp.connect(smtpserver)
+         # TODO: We may need this.  Leaving for now.
+         ## Bits and pieces for authenticated smtp.
+         #smtp.starttls()
+         #smtp.login(sender,password)
+
+         #SMTP.sendmail(from_addr, to_addrs, msg, mail_options=(), rcpt_options=())
+         smtp.sendmail(sender,['cjorr@purdue.edu'], msg.as_string())
+         smtp.sendmail(sender, addr, msg.as_string())
```

Comment:
if the `smtp.sendmail()` accepts the email addresses as a list-only, the second line might fail, as the `addr='snews2-test-ahabig@d.umn.edu'` single string. If this is the case, we can just return `addr=['snews2-test-ahabig@d.umn.edu']`

Reply:
smtp.sendmail() DOES take a list for the to_addrs arg.  If a single string is passed to it, it will treat it as a list with 1 address.
So, I think we're good, although I suppose there's wisdom in future proofing and returning a list anyhow.

(Ref: line 814 in smtplib.py, from Python)

Reply:
On second look, this we need to pass a single string to _smtp_sender, as the code sits now due to how the envelope is assembled in the helper function.